recipes/mysql/build.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+# this script is based off the homebrew package:
+# https://github.com/Homebrew/homebrew-core/blob/master/Formula/mysql.rb
+
+mkdir -p build
+cd build
+
+# -DINSTALL_* are relatiove to -DCMAKE_INSTALL_PREFIX
+mkdir -p ${PREFIX}/mysql
+cmake \
+    -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+    -DMYSQL_DATADIR=${PREFIX}/mysql/datadir \
+    -DINSTALL_INCLUDEDIR=include/mysql \
+    -DINSTALL_MANDIR=share/man \
+    -DINSTALL_DOCDIR=share/doc/mysql \
+    -DINSTALL_DOCREADMEDIR=mysql \
+    -DINSTALL_INFODIR=share/info \
+    -DINSTALL_MYSQLSHAREDIR=share/mysql \
+    -DINSTALL_SUPPORTFILESDIR=mysql/support-files \
+    -DINSTALL_SCRIPTDIR=mysql/scripts \
+    -DSYSCONFDIR=${PREFIX}/etc \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_FIND_FRAMEWORK=LAST \
+    -DCMAKE_VERBOSE_MAKEFILE=OFF \
+    -Wno-dev \
+    -DWITH_UNIT_TESTS=OFF \
+    -DDEFAULT_CHARSET=utf8 \
+    -DDEFAULT_COLLATION=utf8_general_ci \
+    -DCOMPILATION_COMMENT=conda-forge \
+    -DWITH_SSL=bundled \
+    -DWITH_EDITLINE=bundled \
+    -DWITH_READLINE=bundled \
+    -DWITH_BOOST=boost \
```

Comment:
Is it possible not to bundle these things? We include them in `conda-forge`. So it would be nice to depend on them instead.

Reply:
Bundling makes shipping this all easier. I tried to not to bundle and it was hard.

Reply:
Could you please provide some more details on what you mean? Ideally shipping will just be installing with `conda`. What am I missing?

Reply:
Sorry. I meant that compiling mysql against the conda shipped packages is a pain.

Reply:
Did you have the `toolchain` package as a build requirement and pass `-DCMAKE_PREFIX_PATH="$PREFIX"` to `cmake`? That's usually enough....

Reply:
I had the package but I did not know about the extra cmake flag. I am happy to try but I am still curious why it is preferable to use the conda forge package. Letting MySQL bundle the code it needs eliminates dependencies and leads to a more robust build. 

Thoughts?

Reply:
It also introduces extra copies of the libraries on the system, makes the package bigger, and could conceivably lead to problems (maybe? not sure how mysql does it) if someone has a program that links to mysql and also a different version of readline.

Reply:
Everything @dougalsutherland said is correct. Would add more generally that the purpose of adding a package to `conda-forge` is to leverage a well maintained ecosystem of dependencies to lessen the burden of any one maintainer.

For instance, if `mysql` builds `boost` (as an example) in some suboptimal way, then you are on the hook as the `mysql` maintainer for fixing that bundled version somehow. However if you use `conda-forge`'s `boost`, you can lean on the `boost` maintainers here for help. This lessens the amount of working that you are having to do maintaining this package and helps shift the knowledge around in a more efficient way. Namely `boost` maintainers hone their expertise and share it with everyone needing help with `boost`. IOW you just need to focus on making `mysql` work right.

Reply:
Oh, one more benefit: builds should be faster and thus use up less CI resources when you don't also have to build `boost`/`readline`/`editline` with each build of `mysql`.

Reply:
FWIW, the bundled libs don't ship extra copies. They must get compiled in or something.

```
$ ls -lah *
-rwxr-xr-x  1 mbecker  staff   4.1M Nov  1 11:00 libmysqlclient.20.dylib
-rw-r--r--  1 mbecker  staff   5.0M Nov  1 10:59 libmysqlclient.a
lrwxr-xr-x  1 mbecker  staff    23B Nov  1 11:00 libmysqlclient.dylib -> libmysqlclient.20.dylib
-rw-r--r--  1 mbecker  staff    35M Nov  1 10:59 libmysqld.a
-rw-r--r--  1 mbecker  staff   9.4K Nov  1 10:59 libmysqlservices.a

pkgconfig:
total 8
drwxr-xr-x  3 mbecker  staff   102B Nov  1 11:27 .
drwxr-xr-x  9 mbecker  staff   306B Nov  1 11:27 ..
-rw-r--r--  1 mbecker  staff   993B Nov  1 11:01 mysqlclient.pc

plugin:
total 7416
drwxr-xr-x  50 mbecker  staff   1.7K Nov  1 11:27 .
drwxr-xr-x   9 mbecker  staff   306B Nov  1 11:27 ..
-rwxr-xr-x   1 mbecker  staff    19K Nov  1 11:01 adt_null.so
-rwxr-xr-x   1 mbecker  staff    13K Nov  1 11:00 auth.so
-rwxr-xr-x   1 mbecker  staff    13K Nov  1 11:01 auth_test_plugin.so
-rwxr-xr-x   1 mbecker  staff    46K Nov  1 11:00 authentication_ldap_sasl_client.so
-rwxr-xr-x   1 mbecker  staff    47K Nov  1 11:01 connection_control.so
-rw-r--r--   1 mbecker  staff   227B Nov  1 10:31 daemon_example.ini
-rwxr-xr-x   1 mbecker  staff   969K Nov  1 11:00 group_replication.so
-rwxr-xr-x   1 mbecker  staff    31K Nov  1 11:00 ha_example.so
-rwxr-xr-x   1 mbecker  staff    89K Nov  1 11:01 keyring_file.so
-rwxr-xr-x   1 mbecker  staff    20K Nov  1 11:01 keyring_udf.so
-rwxr-xr-x   1 mbecker  staff    14K Nov  1 11:00 libdaemon_example.so
-rwxr-xr-x   1 mbecker  staff   8.5K Nov  1 11:00 libtest_framework.so
-rwxr-xr-x   1 mbecker  staff    14K Nov  1 11:00 libtest_services.so
-rwxr-xr-x   1 mbecker  staff    14K Nov  1 11:01 libtest_services_threaded.so
-rwxr-xr-x   1 mbecker  staff    19K Nov  1 11:01 libtest_session_detach.so
-rwxr-xr-x   1 mbecker  staff    14K Nov  1 11:00 libtest_session_in_thd.so
-rwxr-xr-x   1 mbecker  staff    31K Nov  1 11:00 libtest_session_info.so
-rwxr-xr-x   1 mbecker  staff    23K Nov  1 11:00 libtest_sql_2_sessions.so
-rwxr-xr-x   1 mbecker  staff    27K Nov  1 11:00 libtest_sql_all_col_types.so
-rwxr-xr-x   1 mbecker  staff    29K Nov  1 11:01 libtest_sql_cmds_1.so
-rwxr-xr-x   1 mbecker  staff    27K Nov  1 11:00 libtest_sql_commit.so
-rwxr-xr-x   1 mbecker  staff    28K Nov  1 11:00 libtest_sql_complex.so
-rwxr-xr-x   1 mbecker  staff    23K Nov  1 11:00 libtest_sql_errors.so
-rwxr-xr-x   1 mbecker  staff    27K Nov  1 11:00 libtest_sql_lock.so
-rwxr-xr-x   1 mbecker  staff    23K Nov  1 11:00 libtest_sql_processlist.so
-rwxr-xr-x   1 mbecker  staff    24K Nov  1 11:00 libtest_sql_replication.so
-rwxr-xr-x   1 mbecker  staff    19K Nov  1 11:00 libtest_sql_shutdown.so
-rwxr-xr-x   1 mbecker  staff    28K Nov  1 11:00 libtest_sql_sqlmode.so
-rwxr-xr-x   1 mbecker  staff    28K Nov  1 11:00 libtest_sql_stored_procedures_functions.so
-rwxr-xr-x   1 mbecker  staff    28K Nov  1 11:00 libtest_sql_views_triggers.so
-rwxr-xr-x   1 mbecker  staff    14K Nov  1 11:00 libtest_x_sessions_deinit.so
-rwxr-xr-x   1 mbecker  staff    14K Nov  1 11:00 libtest_x_sessions_init.so
-rwxr-xr-x   1 mbecker  staff    13K Nov  1 11:00 locking_service.so
-rwxr-xr-x   1 mbecker  staff    13K Nov  1 11:00 mypluglib.so
-rwxr-xr-x   1 mbecker  staff   8.6K Nov  1 11:00 mysql_no_login.so
-rwxr-xr-x   1 mbecker  staff   1.5M Nov  1 11:01 mysqlx.so
-rwxr-xr-x   1 mbecker  staff   8.3K Nov  1 11:01 qa_auth_client.so
-rwxr-xr-x   1 mbecker  staff    13K Nov  1 11:01 qa_auth_interface.so
-rwxr-xr-x   1 mbecker  staff   8.7K Nov  1 11:00 qa_auth_server.so
-rwxr-xr-x   1 mbecker  staff    26K Nov  1 11:00 replication_observers_example_plugin.so
-rwxr-xr-x   1 mbecker  staff    13K Nov  1 11:01 rewrite_example.so
-rwxr-xr-x   1 mbecker  staff    48K Nov  1 11:00 rewriter.so
-rwxr-xr-x   1 mbecker  staff    56K Nov  1 11:00 semisync_master.so
-rwxr-xr-x   1 mbecker  staff    16K Nov  1 11:00 semisync_slave.so
-rwxr-xr-x   1 mbecker  staff    13K Nov  1 11:00 test_security_context.so
-rwxr-xr-x   1 mbecker  staff   8.7K Nov  1 11:00 test_udf_services.so
-rwxr-xr-x   1 mbecker  staff    43K Nov  1 11:00 validate_password.so
-rwxr-xr-x   1 mbecker  staff    47K Nov  1 11:00 version_token.so
```

Reply:
Sounds good all. I will work to get it to compile against the shipped packages.

Reply:
Yeah, that sounds like the best thing to do for boost.